### ISSUE TITLE
Cleanroomloggerasl

### ISSUE THREAD

**Issue by navisingh** [emaloney/CleanroomLogger#42]
Why are there two frameworks, cleanroomlogger and cleanroomloggerAsl?  Can they be combined into one?  The way the project is set up creates problems for Jenkins with Carthage

**Comment by emaloney** [comment]
Hi Navi,

There are two reasons [CleanroomASL](https://github.com/emaloney/CleanroomASL) is a separate dependency:

1. The Apple System Log is an Apple-specific subsystem that's only guaranteed to be available on Apple operating systems; CleanroomLogger aims to be supported on any platform Swift can run on.
2. A Swift-based API for the Apple System Log is quite useful outside of CleanroomLogger. For example, CleanroomASL supports querying the ASL for log entries; unlike CleanroomLogger which is write-only, CleanroomASL lets you read logs too.

The benefit of the first point isn't realized yet the native Swift dependency manager (the in-progress Swift Package Manager) effectively makes it impossible to support CleanroomLogger on anything other than OS X for now. But CleanroomLogger is already built in such a way that it should be simple to support once it becomes possible.

What sort of Jenkins/Carthage problem are you having? Although we moved off Carthage and went to straight-up git submodules, we used to use CleanroomLogger in conjunction with Jenkins/Carthage without an issue.

E.
